app/src/main/java/app/gamenative/utils/ContainerConfigIO.kt

```diff
@@ -0,0 +1,488 @@
+                put("fexcoreVersion", container.fexCoreVersion)
+                put("inputType", container.inputType)
+                put("dinputMapperType", container.dinputMapperType.toInt())
+                put("sdlControllerAPI", container.isSdlControllerAPI)
+                put("disableMouseInput", container.isDisableMouseInput)
+                put("touchscreenMode", container.isTouchscreenMode)
+                put("useDRI3", container.isUseDRI3)
+                put("emulateKeyboardMouse", container.isEmulateKeyboardMouse)
+                put("forceDlc", container.isForceDlc)
+                
+                // Additional Container-specific fields not in ContainerData
+                put("primaryController", container.primaryController)
+                put("lc_all", container.getExtra("lc_all", "en_US.utf8"))
+                put("inputType", container.inputType)
+                
+                // Include MIDI sound font if set
+                if (container.midiSoundFont.isNotEmpty()) {
+                    put("midiSoundFont", container.midiSoundFont)
+                }
+                
+                // Include controller mapping if set
+                val controllerMapping = container.getExtra("controllerMapping", "")
+                if (controllerMapping.isNotEmpty()) {
+                    put("controllerMapping", controllerMapping)
+                }
+                
+                // Include controller emulation bindings if set
+                val emulationBindings = container.getExtra("controllerEmulationBindings")
+                if (emulationBindings.isNotEmpty()) {
+                    put("controllerEmulationBindings", JSONObject(emulationBindings))
+                }
+            }
+            
+            put("config", config)
+        }
+        
+        return configData.toString(2) // Pretty print with 2-space indent
+    }
+    
+    /**
+     * Imports a container configuration from a JSON file.
+     * 
+     * @param context Android context
+     * @param sourceUri Document URI to read from (from SAF)
+     * @return ContainerData object if import succeeded, null otherwise
+     */
+    fun importFromFile(context: Context, sourceUri: Uri): ContainerData? {
+        return try {
+            val jsonString = context.contentResolver.openInputStream(sourceUri)?.use { inputStream ->
+                inputStream.bufferedReader().use { it.readText() }
+            } ?: return null
+            
+            importFromJson(jsonString)
+        } catch (e: Exception) {
+            Timber.e(e, "[ContainerConfigIO]: Failed to import config from file")
+            null
+        }
+    }
+    
+    /**
+     * Imports a container configuration from a JSON string.
+     * 
+     * @param jsonString JSON string containing the container config
+     * @return ContainerData object if import succeeded, null otherwise
+     */
+    fun importFromJson(jsonString: String): ContainerData? {
```

Comment:
As a general question, is it worth putting in some input sanitation here?  Might not be required, but worth asking.